src/test/RecipeCard.test.js

```diff
@@ -0,0 +1,128 @@
+        )
+
+        await waitFor(() => screen.getByText('recipe test'))
+        expect(screen.queryByLabelText('Recipe name')).toBeNull()
+        fireEvent.click(screen.getByRole('button', { name: 'Edit recipe' }))
+        expect(screen.getByLabelText('Recipe name')).toHaveValue('recipe test')
+        expect(screen.getByLabelText('Recipe description')).toHaveValue(
+            'my recipe test description'
+        )
+        expect(
+            screen.getByPlaceholderText('New Ingredient')
+        ).toBeInTheDocument()
+        expect(screen.getByText('test ingredient')).toBeInTheDocument()
+    })
+    it('should apply changes when a recipe is edited', async () => {
+        axios.get.mockResolvedValue(recipeMock)
+        const mockedPatchRecipe = jest.fn(() => ({
+            status: 200,
+        }))
+        axios.patch = mockedPatchRecipe
+
+        render(
+            <BrowserRouter>
+                <RecipeCard />
+            </BrowserRouter>
+        )
+
+        await waitFor(() => screen.getByText('recipe test'))
+        userEvent.click(screen.getByRole('button', { name: 'Edit recipe' }))
+        userEvent.type(screen.getByLabelText('Recipe name'), ' edited')
+        userEvent.type(screen.getByLabelText('Recipe description'), ' edited')
+        expect(screen.queryByText('test ingredient')).toBeInTheDocument()
+        userEvent.click(screen.getByRole('button', { name: 'Delete' }))
+        expect(screen.queryByText('test ingredient')).not.toBeInTheDocument()
+        userEvent.type(
+            screen.getByPlaceholderText('New Ingredient'),
+            'new yummy ingredient'
+        )
+        userEvent.click(screen.getByRole('button', { name: 'Add Ingredient' }))
+        await waitFor(() => {
+            screen.getByRole('listitem')
+        })
+        userEvent.click(screen.getByRole('button', { name: 'Save changes' }))
+        await waitFor(() => {
+            expect(mockedPatchRecipe).toHaveBeenCalledWith(
+                'http://localhost:8000/recipes/1/',
+                {
+                    id: 1,
+                    name: 'recipe test edited',
+                    description: 'my recipe test description edited',
+                    ingredients: [{ name: 'new yummy ingredient' }],
+                }
+            )
+        })
+    })
+
+    it('should call the delete api when a recipe is deleted', async () => {
+        axios.get.mockResolvedValue(recipeMock)
+        const mockedDeleteRecipe = jest.fn(() => ({
+            status: 204,
+        }))
+        axios.delete = mockedDeleteRecipe
+
+        render(
+            <BrowserRouter>
+                <RecipeCard />
+            </BrowserRouter>
+        )
+
+        await waitFor(() =>
+            screen.getByRole('button', { name: 'Delete recipe' })
+        )
+        fireEvent.click(screen.getByRole('button', { name: 'Delete recipe' }))
+
+        await waitFor(() => {
+            expect(mockedDeleteRecipe).toHaveBeenCalled()
+        })
```

Comment:
after the deletion is completed there are some changes in the UI, if you don't add tests from them the testing system is going to complain with some like _'you are calling setState after the test completes'_